dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -906,6 +950,355 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
+     * Add the workflow labels as keywords to the deposition metadata
+     * @param depositMetadata Metadata for the workflow version
+     * @param workflow    workflow for which DOI is registered
+     */
+    private void setMetadataKeywords(DepositMetadata depositMetadata, Workflow workflow) {
+        // Use the Dockstore workflow labels as Zenodo free form keywords for this deposition.
+        Set<Label> workflowLabels = workflow.getLabels();
+        Iterator labelIter = workflowLabels.iterator();
+        List<String> labelList = new ArrayList<>();
+        while (labelIter.hasNext()) {
+            String label = ((Label)labelIter.next()).getValue();
+            labelList.add(label);
+        }
+        depositMetadata.setKeywords(labelList);
+    }
+
+    /**
+     * Add the workflow aliases as related identifiers to the deposition metadata
+     * @param depositMetadata Metadata for the workflow version
+     * @param workflow    workflow for which DOI is registered
+     */
+    private void setMetadataRelatedIdentifiers(DepositMetadata depositMetadata, Workflow workflow) {
+        // Get the aliases for this workflow and add them to the deposit
+        // The alias must be a format supported by Zenodo such as
+        // DOI, Handle, ARK...URNs and URLs
+        // See http://developers.zenodo.org/#representation 'related identifiers'
+        Set<String> workflowAliases = workflow.getAliases().keySet();
+        Iterator iter = workflowAliases.iterator();
+        List<RelatedIdentifier> aliasList = new ArrayList<RelatedIdentifier>();
+        while (iter.hasNext()) {
+            RelatedIdentifier relatedIdentifier = new RelatedIdentifier();
+            String workflowAlias = ((String)iter.next());
+            relatedIdentifier.setIdentifier(workflowAlias);
+            relatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
+            aliasList.add(relatedIdentifier);
+        }
+
+        // TODO Use stream map collect to list
+        // List<RelatedIdentifier> aliasList = workflowAliases.stream().map(s -> {
+        //     RelatedIdentifier relatedIdentifier = new RelatedIdentifier();
+        //     relatedIdentifier.setIdentifier(s);
+        //     relatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
+        //     //aliasList.add(relatedIdentifier);
+        //     return relatedIdentifier;
+        // }).collect(Collectors.toList());
+
+
+        depositMetadata.setRelatedIdentifiers(aliasList);
+    }
+
+    /**
+     * Add the workflow author as creator to the deposition metadata
+     * @param depositMetadata Metadata for the workflow version
+     * @param workflow    workflow for which DOI is registered
+     */
+    private void setMetadataCreator(DepositMetadata depositMetadata, Workflow workflow) {
+        String wfAuthor = workflow.getAuthor();
+        String authorStr = (wfAuthor == null || wfAuthor.isEmpty()) ? "unknown creator" : workflow.getAuthor();
```

Comment:
try `String authorStr = Optional.ofNullable(wfAuthor).orElse("unknown creator");`

Reply:
👍 I like and did not know that technique.

OTOH it won't catch the empty string, and I think Walt said Zenodo may require a non-empty string.

Reply:
done

Reply:
hmm...looks like https://stackoverflow.com/a/31412469/4484759 is wrong :(

`StringUtils.defaultIfEmpty(wfAuthor, "unknown creator")` should work though